src/main/java/org/codehaus/mojo/versions/UpdateChildModulesMojo.java

```diff
@@ -79,24 +77,20 @@ public void execute()
         {
             final Map<String, Model> reactor = PomHelper.getReactorModels( getProject(), getLog() );
             List<String> order = new ArrayList<>( reactor.keySet() );
-            Collections.sort( order, new Comparator<String>()
-            {
-                public int compare( String o1, String o2 )
+            order.sort( ( o1, o2 ) -> {
+                Model m1 = reactor.get( o1 );
+                Model m2 = reactor.get( o2 );
+                int d1 = PomHelper.getReactorParentCount( reactor, m1 );
+                int d2 = PomHelper.getReactorParentCount( reactor, m2 );
+                if ( d1 < d2 )
                 {
-                    Model m1 = reactor.get( o1 );
-                    Model m2 = reactor.get( o2 );
-                    int d1 = PomHelper.getReactorParentCount( reactor, m1 );
-                    int d2 = PomHelper.getReactorParentCount( reactor, m2 );
-                    if ( d1 < d2 )
-                    {
-                        return -1;
-                    }
-                    else if ( d1 > d2 )
-                    {
-                        return 1;
-                    }
-                    return 0;
+                    return -1;
+                }
+                else if ( d1 > d2 )
+                {
+                    return 1;
                 }
+                return 0;
             } );
```

Comment:
Refactor out this lambda. Lambdas should not exceed a few lines and not have more than one return. Make this a new comparator, preferably a `static final` one if this method is being called frequently.

Reply:
this comparator need a local variable

Reply:
Yeah, so make it a method which gets three instead of two parameters. 🤷🏻‍♂️
can do it afterwards if you like.